backend/mutations.py

```diff
@@ -1,58 +1,78 @@
 import graphene
-
-from models import User, UserMetrics, Portfolio, Settings, Achievement, Group
+from bson import ObjectId
+from models import User, UserMetrics, Portfolio, Settings, Achievement, Group, Artwork, ArtworkMetrics
 from api_types import UserType, UserMetricsType, PortfolioType, SettingsType, AchievementType, GroupType
+from api_types import ArtworkType
 
 '''
-List of possible mutations:
-    create_user
-    update_user - need to add update metrics and settings
-    delete_user
+Need to research more into mutation arguments, but I think how I have them right now is how it should be
+List of possible mutations to make:
 
     create_group - todo
     update_group - todo
-    delete_group - todo, when should a group be deleted?
+    delete_group - todo
 
-    
+    create_achievement - todo
+    update_achievement - todo
+    remove_achievement - todo
+
+    incrementing metrics mutations (users and artworks) - todo    
 '''
 
+# NOTE: reference fields need ObjectId-typed input
+# NOTE: Collection methods - Remove deletes everything in the list, pop removes by index
+# Idea, might wanna store the graphql id in mongodb or vise versa somehow
+
+class UserSettingsInput(graphene.InputObjectType):
+    autoAddToGroupPortfolio = graphene.Boolean()
 
+class UserMetricsInput(graphene.InputObjectType):
+    works_visited = graphene.Int()
+    works_found = graphene.Int()
+    cities_visited = graphene.Int()
+    posts_written = graphene.Int()
 
 class UserInput(graphene.InputObjectType):
-    # id = graphene.ID()
-    name = graphene.String(required=True)
+    # since this class is generalized for all updating and creating, won't make any attributes required
+    id = graphene.String()
+    name = graphene.String()
     bio = graphene.String()
+    profile_pic = graphene.String()   # NOTE: will assume base64 encoded string
+    metrics = graphene.InputField(UserMetricsInput)
+    achievement = graphene.String()     # achievement id to add to user's achievements
+    art_to_add = graphene.String()    # artwork id to add to portfolio
+    art_to_remove = graphene.String()     # artwork id to remove from portfolio
+    group = graphene.String()         # group id to add to user's groups
+    settings = graphene.InputField(UserSettingsInput)
```

Comment:
Maybe these should be separate mutations?

Reply:
Yeah I just wanted to get everything to work first, but I can focus on breaking it up as well.